### ISSUE TITLE
Update error in Virtual Box

### ISSUE THREAD

**Issue by LupusPluvia** [getumbrel/umbrel#1884]
Hello, I have the following problem. In my virtual box. When I want to update the umbrel OS, what is shown in the photo always appears. And then I have to reinstall everything. Can someone help me? The reason for the change is that the Pi 4 is too weak in terms of performance.

![20240729_202144](https://github.com/user-attachments/assets/73c851aa-c762-4632-ac71-d5950c46edab)

**Comment by LupusPluvia** [comment]
If I just restart it the same thing happens.[
![20240729_214014](https://github.com/user-attachments/assets/90649438-afe0-4c31-a2cd-836111b1e2a9)
](url)

**Comment by a4004** [comment]
Hey @LupusPluvia 👋 

It looks like there's a problem mounting the data volume for some reason. It's possible that you haven't allocated enough system resources to the virtual machine hence why you might be experiencing the issues.

Since you mentioned, you've not transferred anything the easiest and quickest thing to do in this case would be to completely reinstall umbrelOS - ideally on a new virtual machine using the latest version of the ISO.

I recommend allocating your virtual machine the following system resources to ensure you can get a decent experience:
- **8 GB memory** at least (4 GB is the absolute minimum, but do not use a Bitcoin Node or anything dependent on it)
- **50 GB of storage** at least for a decent experience (give it more if you will store data)
- **4 vCPU's** at least
- U**EFI** boot mode

Let me know if you need any further help! 🫂

**Comment by LupusPluvia** [comment]
![Uploading 
Screenshot_20240801_170159_TeamViewer.jpg…]()

What I have to say is that it is not started from a virtual disk.

**Comment by LupusPluvia** [comment]
![Screenshot_20240801_170159_TeamViewer](https://github.com/user-attachments/assets/bbfdd364-e933-41b6-bf19-64aaaa1c2c3d)

**Comment by LupusPluvia** [comment]
But you gave me an idea. I could create a virtual disk on the HDD with the same size. But I have one more question: can the data from the Pi4 be migrated there too?

**Comment by LupusPluvia** [comment]
Or is there a possibility like with the pi4 that two storage systems can be used. If I do that, only one storage medium is used and that is where it is installed and the second disk will be ignored.

**Comment by a4004** [comment]
> But you gave me an idea. I could create a virtual disk on the HDD with the same size. But I have one more question: can the data from the Pi4 be migrated there too?

Unfortunately no, the Umbrel Migration Assistant is only available on the Umbrel Home as of v1.2.2 as you can see here in this screenshot from my virtual machine running umbrelOS:

![image](https://github.com/user-attachments/assets/788506ae-a0e9-4b8b-bcdc-32b614b80214)


> Or is there a possibility like with the pi4 that two storage systems can be used. If I do that, only one storage medium is used and that is where it is installed and the second disk will be ignored.

At the moment, there is no official support for external storage drives on umbrelOS. 

On the Raspberry Pi variant of the system, there is support for a single external storage drive in order to allow users to have more space beyond the size of typically small SD cards - which are commonly used to boot an OS on the Raspberry Pi's. 

That being said, you can mount additional storage drives on umbrelOS using [this unofficial guide](https://community.umbrel.com/t/unofficial-guide-how-to-add-more-storage/18397); however, keep in mind it's only a temporary workaround and full support will be coming in a future update.

**Comment by a4004** [comment]
The x86 version of umbrelOS uses a single drive for the OS and data so if you erase the main drive, you will lose all your data. If you set-up a second drive using the guide, then anything stored on the second drive will be safe, but it's not properly integrated yet.

**Comment by LupusPluvia** [comment]
As I understand it now, if I have to reinstall the system because the OS system is broken, everything is gone. Is there a way to access the virtual HDD if something like that happens... because it can't be set up as a normal physical HDD without having the problem mentioned above.

**Comment by a4004** [comment]
Yes, you can open the virtual machine settings in VirtualBox, find the hard drive and look at where the `.vdi` file is stored.

Then go to that path and copy the VDI file to another location and start recovering.

If you ever need to recover it you can either mount it in a new VM running any Linux distro, or you can convert it to another format and mount it in your file manager to access the files.

**Comment by LupusPluvia** [comment]
@a4004 The system runs as a vmdk and with dynamic memory expansion. I noticed that if I were to enlarge the vmdk, the OS would not boot and I would have to reinstall it. Mounting the disk worked so far, but it's not how I wanted it. I wanted the OS to start with all the data from the pi4, since the migration doesn't work or isn't working yet. Is there a solution how else I could do this.

**Comment by LupusPluvia** [comment]
And if everything is to work, it should all be on a 4tb×2 m2. nvme in raid 1, that was my plan anyway. As long as I have everything together.

**Comment by LupusPluvia** [comment]
@a4004 If I were to transfer everything from the Pi with sfpt, which directory would it have to go to, as it is probably structured differently to the Pi, or would it all go in mnt. or directly into the umbrel directory?

**Comment by a4004** [comment]
The data directory on umbrelOS is at `/home/umbrel/umbrel/data`.

Also, from my own experience I recommend not using any dynamic storage or memory feature in your virtual machine as it may not play nicely with the system.

**Comment by LupusPluvia** [comment]
@a4004  What would need to be transferred or backed up from the directory?... because not everything is possible. could you make me a list.

**Comment by LupusPluvia** [comment]
What would you use LND or C leitning have used lnd before.

**Comment by jirielgr** [comment]
> If I just restart it the same thing happens.[ ![20240729_214014](https://private-user-images.githubusercontent.com/15277011/353210729-90649438-afe0-4c31-a2cd-836111b1e2a9.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUxMDc4NjQsIm5iZiI6MTcyNTEwNzU2NCwicGF0aCI6Ii8xNTI3NzAxMS8zNTMyMTA3MjktOTA2NDk0MzgtYWZlMC00YzMxLWEyY2QtODM2MTExYjFlMmE5LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMxVDEyMzI0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjMTFiODRhNmM1MzYwNzVlOTE2ZWQ4MDBjMGIzMWFiODEyNjlmZWE5YmMxN2UxYWY5NDY3MmEwYWExZDFiYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OUoNfkg2rCBLB9OvRM4M0wC2ZPsngKAP3m3RWxn3Vro) ](url)

This could be fixed by this https://community.umbrel.com/t/boot-problems-failed-failed-to-mount-data-mount-data/17207/4
Boot from live distro, mount the share and "fix" the fstab --> reboot and wait :)
